### PR TITLE
[PLAT-6019] Update node e2e tests to Maze-runner v4

### PR DESCRIPTION
**Pull request by Cawllec**
## Goal

Enables the use of Maze-runner v4 in node end-to-end tests, which includes improvements such as separate endpoints that should reduce the chance of flakes.

Due to debug commits this PR should be squash upon merging.

### CODE REVIEW

File: .buildkite/node-pipeline.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ steps:
         run: node-maze-runner
         use-aliases: true
         verbose: true
-        command: ["-e", "koa.feature", "-e", "koa-1x.feature", "-e", "webpack.feature"]
+        command: ["--bind-address=0.0.0.0", "-e", "koa.feature", "-e", "koa-1x.feature", "-e", "webpack.feature"]
```
Comment (twometresteve) [inline .buildkite/node-pipeline.yml:28]: Just to note that this shouldn't be need in the next Maze Runner release (but no need to hold up the PR for that).

File: test/node/Gemfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+source 'https://rubygems.org'
+
+gem 'cocoapods'
```
Comment (twometresteve) [inline test/node/Gemfile:3]: Is this needed for Node?

File: test/node/Gemfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+# A reference to Maze Runner is only needed for running tests locally and if committed it must be
+# portable for CI, e.g. a specific release.  However, leaving it commented out would mean quicker CI.
+gem 'bugsnag-maze-runner', git: 'https://github.com/bugsnag/maze-runner', tag: 'v4.11.3'
+
+# Locally, you can run against Maze Runner branches and uncommitted changes:
+#gem 'bugsnag-maze-runner', path: '../maze-runner'
```
Comment (twometresteve) [inline test/node/Gemfile:10]:
```suggestion
#gem 'bugsnag-maze-runner', path: '../../../maze-runner'
```

**Review by twometresteve — APPROVED** [review]
Looks good, just a couple of small potential tweaks.

**Review by imjoehaines — APPROVED** [review]

**Pull request merged**